### PR TITLE
Better diagnostics on ISOTXS compare 

### PR DESCRIPTION
**Pull request by onufer**
## What is the change?

This makes the isotxs diffs useable. we had this in the past but at some point it got lost. what you get now is unclear (right side). What you get after this PR is headers that tell you what is not equal
image

## Why is the change being made?

because at some point the useful messages were deleted and the compare is not currently useful.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [ ] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Review by john-science — APPROVED** [review]

File: armi/nuclearDataIO/cccc/isotxs.py [hunk 1]
```diff
@@ -123,7 +124,10 @@ def compare(lib1, lib2, tolerance=0.0, verbose=False):
                 runLog.warning(warning.format(nuc2, 2, 1))
             equal = False
             continue
-        equal &= compareNuclideXS(nuc1, nuc2, tolerance, verbose)
+        nucEqual = compareNuclideXS(nuc1, nuc2, tolerance, verbose)
```
Comment (jakehader) [inline armi/nuclearDataIO/cccc/isotxs.py:127]: Could consider expanding in this compareNuclideXS function to write out all differences or plot them for further assessments.

**Review by aaronjamesreynolds — APPROVED** [review]
This looks good to me. This slight changes greatly improves the usefulness of this comparison tool.

**Comment by onufer** [comment]
i pushed up a change that will not print anything for nuclides that are equal i think this is good since it will print less lines
image

File: armi/nuclearDataIO/cccc/isotxs.py (hunk 1 above)
Comment (onufer) [reply armi/nuclearDataIO/cccc/isotxs.py]: verbose prints the matricies. there are often hundreds of differences and hundreds of plots im not sure are that benifical.

**Review by aaronjamesreynolds — APPROVED** [review]
The new additions make it so that ARMI only identifies XSs that differ between two ISOTXS libraries. This makes the diff easier to read as it isn't crowded with information about XSs that agree.

A confirmation message is still printed out if all XSs in each library are identical, which is a good explicit signal for the user to have.

**Comment by john-science** [comment]
@onufer  Are you hoping this PR will make it into the (imminent) ARMI 0.5.0 release?

**Comment by opotowsky** [comment]
@john-science I think this is safe to merge. I do believe it's meant to get into the 0.5 release

**Pull request merged**